added palette global (#3984)
slint-ui/slint: docs/reference/src/language/builtins/globals.md

<!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
# Builtin Global Singletons

## `Palette`

> `Palette` give access to different brushes that can be used to to create custom widgets that matches the colors of
> the selected style e.g. fluent, cupertino, material or qt.

Reviewer: ```suggestion
Use `Palette` to create custom widgets that match the colors of
the selected style e.g. fluent, cupertino, material, or qt.
```

### Properties

> -   **`background`** (_out_ _brush_): Defines the default background brush.

Reviewer: ```suggestion
-   **`background`** (_out_ _brush_): Defines the default background brush. Use this if none of the more specialised background brushes apply.
```

<!-- Copyright © SixtyFPS GmbH <info@slint.dev> ; SPDX-License-Identifier: MIT -->
# Builtin Global Singletons

## `Palette`

`Palette` give access to different brushes that can be used to to create custom widgets that matches the colors of
the selected style e.g. fluent, cupertino, material or qt.

### Properties

-   **`background`** (_out_ _brush_): Defines the background of the window.
-   **`on-background`** (_out_ _brush_): Defines the text color used on `background`.
-   **`accent`** (_out_ _brush_): Defines the color of highlighted elements e.g. a primary button.
-   **`on-accent`** (_out_ _brush_): Defines the text color used on `accent`.
-   **`surface`** (_out_ _brush_): Defines the main widget background color e.g. for button.
-   **`on-surface`** (_out_ _brush_): Defines the text color used on `surface`.
-   **`border`** (_out_ _brush_): Defines the border color of widgets.

> -   **`selection`** (_out_ _brush_): Defines the text selection color.

Author: It's not the same.

Selection is only the text selection brush. Accent is e.g. the background brush of a primary button or the brush for the slider thumb.

> -   **`selection`** (_out_ _brush_): Defines the text selection color.

Author: And on-selection is depending on the background of the widget. so it could be e.g. on-surface or on-background.

the selected style e.g. fluent, cupertino, material, or qt.

### Properties

-   **`background`** (_out_ _brush_): Defines the default background brush. Use this if none of the more specialised background brushes apply.
-   **`foreground`** (_out_ _brush_): Defines the foreground brush that is used for content that is displayed on `background` brush.
-   **`alternate-background`** (_out_ _brush_): Defines an alternate background brush.
-   **`alternate-foreground`** (_out_ _brush_): Defines the foreground brush that is used for content that is displayed on `alternate-background` brush.
-   **`control-background`** (_out_ _brush_): Defines the default background brush for controls, such as push buttons, combo boxes, etc.
-   **`control-foreground`** (_out_ _brush_): Defines the foreground brush that is used for content that is displayed on `control-background` brush.
-   **`accent-background`** (_out_ _brush_): Defines the background brush for highlighted controls such as primary buttons.
-   **`accent-foreground`** (_out_ _brush_): Defines the foreground brush that is used for content that is displayed on `accent-background` brush.

> -   **`selection`** (_out_ _brush_): Defines the background brush that is used to highlight a selection such as a text selection.

Reviewer: Oh, I just realize: The proposal we discussed had `selection-background` and `selection-foreground`. Could you briefly comment on why you decided to just go for `selection` instead of two properties?

Author: ADDED

## `Palette`

Use `Palette` to create custom widgets that match the colors of
the selected style e.g. fluent, cupertino, material, or qt.

### Properties

-   **`background`** (_out_ _brush_): Defines the default background brush. Use this if none of the more specialised background brushes apply.
-   **`foreground`** (_out_ _brush_): Defines the foreground brush that is used for content that is displayed on `background` brush.

> -   **`alternate-background`** (_out_ _brush_): Defines an alternate background brush.

Reviewer: Could you include an example where this is used in current styles, to help users decide whether to go for `background` or `alternate-background`?

Author: done
